components/raftstore/src/store/fsm/apply.rs

```diff
@@ -3538,38 +3565,67 @@ where
         msgs: &mut Vec<Msg<EK>>,
     ) {
         let mut drainer = msgs.drain(..);
+        let mut batch_apply = None;
```

Comment:
What is the benefit to batch apply messages of the same message batch? I think the writes are already batched.

Reply:
Found that `WriteBatch` is append-only, and each `finish_for` call will append a `ApplyState`. To reduce write of `ApplyState` of the same region we can also cache the `ApplyState` with `HashMap` and merge them to the `WriteBatch` before it is committed to rocksdb.

Reply:
I change this logic, it doesn't need to use a `HashMap`. The `ApplyState` will be written only when
1. no next msg or next msg is not apply msg
2. yield

Reply:
I remove this optimization to simplify this PR.